library/blivet.py

```diff
@@ -0,0 +1,606 @@
+#!/usr/bin/python
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: blivet
+
+short_description: Module for management of linux block device stacks
+
+version_added: "2.5"
+
+description:
+    - "Module configures storage pools and volumes to match the state specified
+       in input parameters. It does not do any management of /etc/fstab entries."
+
+options:
+    pools:
+        description:
+            - list of dicts describing pools
+    volumes:
+        description:
+            - list of dicts describing volumes
+    use_partitions:
+        description:
+            - boolean indicating whether to create partitions on disks for pool backing devices
+    disklabel_type:
+        description:
+            - disklabel type string (eg: 'gpt') to use, overriding the built-in logic in blivet
+
+author:
+    - David Lehman (dlehman@redhat.com)
+'''
+
+EXAMPLES = '''
+
+- name: Manage devices
+  blivet:
+    pools:
+        - name: "{{ app_pool }}"
+          disks: ["sdd", "sde"]
+          volumes:
+            - name: shared
+              size: "10 GiB"
+              mount_point: /opt/{{ app_pool }}/shared
+            - name: web
+              size: 8g
+              mount_point: /opt/{{ app_pool }}/web
+    volumes:
+        - name: whole_disk1
+          disks: ['sdc']
+          mount_point: /whole_disk1
+          fs_type: ext4
+          mount_options: journal_checksum,async,noexec
+'''
+
+RETURN = '''
+actions:
+    description: list of dicts describing actions taken
+    type: list of dict
+leaves:
+    description: list of paths to leaf devices
+    type: list of str
+mounts:
+    description: list of dicts describing mounts to set up
+    type: list of dict
+pools:
+    description: list of dicts describing the pools w/ device path for each volume
+    type: list of dict
+volumes:
+    description: list of dicts describing the volumes w/ device path for each
+    type: list of dict
+'''
+
+from blivet import Blivet
```

Comment:
this does not work on RHEL 7. I think you have to use `blivet3` since that's what you are installing on https://github.com/linux-system-roles/storage/blob/9bf8d5af3d8351f128eb73b9b9033c1311cf0eb7/tasks/main-blivet.yml#L9 
https://github.com/linux-system-roles/storage/blob/9bf8d5af3d8351f128eb73b9b9033c1311cf0eb7/vars/main.yml#L1